core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1104,35 +1104,43 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     val responseTopics = authorizedRequestInfo.map { topic =>
       val responsePartitions = topic.partitions.asScala.map { partition =>
-        val topicPartition = new TopicPartition(topic.name, partition.partitionIndex)
-
-        try {
-          val offsets = replicaManager.legacyFetchOffsetsForTimestamp(
-            topicPartition = topicPartition,
-            timestamp = partition.timestamp,
-            maxNumOffsets = partition.maxNumOffsets,
-            isFromConsumer = offsetRequest.replicaId == ListOffsetsRequest.CONSUMER_REPLICA_ID,
-            fetchOnlyFromLeader = offsetRequest.replicaId != ListOffsetsRequest.DEBUGGING_REPLICA_ID)
+        if (partition.timestamp() < ListOffsetsRequest.EARLIEST_TIMESTAMP) {
```

Comment:
Should we remove the code related to `EARLIEST_LOCAL_TIMESTAMP` in `UnifiedLog.legacyFetchOffsetsBefore`?

Reply:
Hi @junrao, thanks for the suggestion. I remove `EARLIEST_LOCAL_TIMESTAMP` case in `UnifiedLog.legacyFetchOffsetsBefore`.